src/components/posts/view-post/PostPage.tsx

```diff
@@ -297,11 +302,12 @@ export async function loadPostOnNextReq({
 
   if (!postId) return return404(context)
 
-  const replyIds = await blockchain.getReplyIdsByPostId(idToBn(postId))
-
-  const ids = replyIds.concat(postId)
-
-  await dispatch(fetchPosts({ api: subsocial, ids, reload: true, eagerLoadHandles: true }))
+  async function getPost() {
+    const replyIds = await blockchain.getReplyIdsByPostId(idToBn(postId!))
+    const ids = replyIds.concat(postId!)
+    await dispatch(fetchPosts({ api: subsocial, ids, reload: true, eagerLoadHandles: true }))
+  }
+  await Promise.all([getPost(), dispatch(fetchBlockedResources({ appId }))])
```

Comment:
The `fetchBlockedResources` has cache? I mean that each post page doesn't load in again, right?

Reply:
no, currently it doesn't have any cache,
but getPost also doesn't use cache right?

and if we want to use cache here, its a bit hard to manage when should you use the cached/new data (when new blocked resource are added)

unless we just want to use time based cache, like redis for 5secs

Reply:
I see, ok lets go without cache

Maybe make sense improve backend query for getting data only for particular post